integration-tests/mongodb-panache-kotlin/src/main/kotlin/io/quarkus/it/mongodb/panache/reactive/book/ReactiveBookEntityResource.kt

```diff
@@ -52,19 +51,19 @@ class ReactiveBookEntityResource {
 
     @POST
     fun addBook(book: ReactiveBookEntity): Uni<Response> {
-        return book.persist().map {
+        return book.persist< ReactiveBookEntity>().map {
             //the ID is populated before sending it to the database
             Response.created(URI.create("/books/entity${book.id}")).build()
         }
     }
 
     @PUT
-    fun updateBook(book: ReactiveBookEntity): Uni<Response> = book.update().map { _ -> Response.accepted().build() }
+    fun updateBook(book: ReactiveBookEntity): Uni<Response> = book.update<ReactiveBookEntity>().map { Response.accepted().build() }
```

Comment:
Huh, this seems suboptimal for a parameter we're not even using… Any idea how we can improve that, @evanchooly ?

Reply:
without a local variable to resolve that type against, you'll need this.  maybe `book.update<*>()` might work but it'd be hard to argue that it's "better."  Less verbose at least.